protos/local/core_interface.proto

```diff
@@ -183,68 +173,48 @@ message WFActivationCompletion {
     }
 }
 
-/// Used to report activity completion to core and to resolve the activity in a workflow activtion
+/// Used to report activity completion to core and to resolve the activity in a workflow activation
 message ActivityResult {
     oneof status {
         ActivityTaskSuccess completed = 1;
-        ActivityTaskCancelation canceled = 2;
-        ActivityTaskFailure failed = 3;
+        ActivityTaskFailure failed = 2;
+        ActivityTaskCancelation canceled = 3;
     }
 }
 
-/// Request cancellation of an activity from a workflow
-message RequestActivityCancellation {
-    string activity_id = 1;
-    string reason = 2;
-}
-
 message CoreCommand {
     oneof variant {
-        temporal.api.query.v1.WorkflowQueryResult respond_to_query = 1;
-        RequestActivityCancellation request_activity_cancellation = 2;
-    }
-}
-
-// Included in successful [WfActivationCompletion]s, indicates what the workflow wishes to do next
-message Command {
-    oneof variant {
-        temporal.api.command.v1.Command api = 1;
-        CoreCommand core = 2;
     }
 }
 
 message WFActivationSuccess {
     // A list of commands to send back to the temporal server
-    repeated Command commands = 1;
-
-    // Other bits from RespondWorkflowTaskCompletedRequest as needed
+    repeated workflow_commands.WorkflowCommand commands = 1;
 }
 
 message WFActivationFailure {
-    temporal.api.enums.v1.WorkflowTaskFailedCause cause = 1;
-    temporal.api.failure.v1.Failure failure = 2;
-
-    // Other bits from RespondWorkflowTaskFailedRequest as needed
+    common.UserCodeFailure failure = 1;
 }
 
 /// Used in ActivityResult to report cancellation
 message ActivityTaskCancelation {
```

Comment:
I prefer to put all of the activity result variants in a [separate file](https://github.com/temporalio/sdk-core/blob/739f20dd249e6b6d3ef95c86f02f6861d2938aeb/protos/local/activity_result.proto).